ext/re2/extconf.rb

```diff
@@ -44,24 +44,35 @@
   abort "You must have re2 installed and specified with --with-re2-dir, please see https://github.com/google/re2/wiki/Install"
 end
 
-# Recent versions of re2 now require a compiler with C++11 support
-checking_for("re2 requires C++11 compiler") do
+TRY_CXXFLAGS = %w[-std=c++20 -std=c++17 -std=c++11 -std=c++0x]
+
+# Recent versions of re2 depend directly on abseil, which requires a
+# compiler with C++14 support (see
+# https://github.com/abseil/abseil-cpp/issues/1127 and
+# https://github.com/abseil/abseil-cpp/issues/1431). However, the
+# `std=c++14` flag doesn't appear to suffice; we need at least
+# `std=c++17`.
+checking_for("re2 requires a C++14 compiler") do
```

Comment:
Is there a way for us to print the relevant message as we check for a C++17 compiler separately from checking for a C++11 compiler (for versions of re2 prior to release 2023-07-01)?

It's also worth noting that the return value of the block is used to determine whether it will print `yes` or `no` when compiling, e.g. when I tried this on my machine with the latest release, I see:

```
checking for re2 requires a C++14 compiler... no
```

Reply:
@mudge Good point, I've improved the messages.